src/components/layout/index.tsx

```diff
@@ -1,29 +1,20 @@
-import Footer from '@/components/footer'
-import { Node } from '@/lib/delegators/Node'
+import { Banner } from '@/components/banner'
 import { isEmpty } from 'lodash'
 export interface LayoutProps {
   children?: React.ReactNode
-  props?: {
-    // eslint-disable-next-line
-    bannerData?: any
-    footerData: object
-  }
+  props?: any
+  bannerData?: any
 }
 
-export default function Layout({ children, props }: LayoutProps) {
+export default function Layout({ props, bannerData, children }: LayoutProps) {
+  if (isEmpty(bannerData)) bannerData = props.bannerData
+
   return (
```

Comment:
This is where we either look for homepage or demo routes: `bannerData` or  slug route: `props.bannerData`